src/libcore/num/mod.rs

```diff
@@ -150,6 +151,101 @@ macro_rules! int_impl {
+        ///
+        /// # Panics
+        ///
+        /// This function will panic if `radix` is smaller than 2 or larger than 36,
+        /// or if `buffer` is too small.
+        /// As a conservative upper bound, `&mut [u8; 128]` is always large enough.
+        ///
+        /// # Safety
+        ///
+        /// `buffer` may be uninitialized.
+        ///
+        /// # Examples
+        ///
+        /// ```
+        /// #![feature(int_to_str)]
+        ///
+        /// assert_eq!((i16::min_value() + 1).to_uppercase_str_radix(&mut [0; 5], 16), "-7FFF")
+        /// ```
+        #[unstable(feature = "int_to_str", issue = /* FIXME */ "0")]
+        pub fn to_uppercase_str_radix(self, buffer: &mut [u8], radix: u32) -> &mut str {
+            fmt::num::SignedToStr::to_str_radix(self as $ActualT, buffer, radix, true)
+        }
+
+        /// Write the representation in a given base in a pre-allocated buffer.
+        ///
+        /// Digits are a subset (depending on `radix`) of `0-9a-z`.
+        ///
+        /// The returned slice starts with a minus sign for negative values
+        /// but contains no leading zero or plus sign,
+        /// and is aligned to the *end* of `buffer`.
+        ///
+        /// # Panics
+        ///
+        /// This function will panic if `radix` is smaller than 2 or larger than 36,
+        /// or if `buffer` is too small.
+        /// As a conservative upper bound, `&mut [u8; 128]` is always large enough.
+        ///
+        /// # Safety
+        ///
+        /// `buffer` may be uninitialized.
+        ///
+        /// # Examples
+        ///
+        /// ```
+        /// #![feature(int_to_str)]
+        ///
+        /// assert_eq!((i16::min_value() + 1).to_lowercase_str_radix(&mut [0; 5], 16), "-7fff")
+        /// ```
+        #[unstable(feature = "int_to_str", issue = /* FIXME */ "0")]
+        pub fn to_lowercase_str_radix(self, buffer: &mut [u8], radix: u32) -> &mut str {
+            fmt::num::SignedToStr::to_str_radix(self as $ActualT, buffer, radix, false)
+        }
+
+        /// Writes the decimal representation in a pre-allocated buffer.
+        ///
+        /// The returned slice starts with a minus sign for negative values
+        /// but contains no leading zero or plus sign,
+        /// and is aligned to the *end* of `buffer`.
+        ///
+        /// # Panics
+        ///
+        /// This function will panic if `buffer` is smaller than [`MAX_STR_LEN`].
+        ///
+        /// [`MAX_STR_LEN`]: #associatedconstant.MAX_STR_LEN
+        ///
+        /// # Safety
+        ///
+        /// `buffer` may be uninitialized.
+        ///
+        /// # Examples
+        ///
+        /// ```
+        /// #![feature(int_to_str)]
+        ///
+        /// assert_eq!(i16::min_value().to_str(&mut [0; i16::MAX_STR_LEN]), "-32768")
+        /// ```
+        #[unstable(feature = "int_to_str", issue = /* FIXME */ "0")]
+        pub fn to_str(self, buffer: &mut [u8]) -> &mut str {
```

Comment:
I am not thrilled about how an inherent method `to_str` (which is almost never what you want) competes in rustdoc searches and autocomplete with a trait method `to_string` (which is almost always what you want). Would it make sense to name this in a scarier way?